Backend/src/main/java/com/harmonia/backend/controller/PublicMessageController.java

```diff
@@ -8,38 +8,74 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+/**
+ * This class represents a REST controller for handling public messages.
+ */
 @RestController
 @RequestMapping("pmessage")
 public class PublicMessageController {
 
+    /**
+     * Autowires an instance of PublicMessageService to be used by this controller.
+     */
     @Autowired
     private PublicMessageService publicMessageService;
 
+    /**
+     * Constructor that sets the PublicMessageService for this controller.
+     *
+     * @param pmService the PublicMessageService instance to be used by this controller.
+     */
     public PublicMessageController(PublicMessageService pmService) {
         this.publicMessageService = pmService;
     }
 
+    /**
+     * Returns a list of all public messages.
+     *
+     * @param dmId (optional) the id of the public message to filter by.
+     * @return an iterable of PublicMessage objects.
+     */
     @GetMapping
     @CrossOrigin
     public Iterable<PublicMessage> listMessages(@RequestParam(name = "pmessageId", required = false) String dmId) {
```

Comment:
Unless you want to do the getById and the list all in the same endpoint, I would remove the @RequestParam(name = "pmessageId", required = false) String dmId or give it a purpose like channelId